templates/vue/src/store/actions.js

```diff
@@ -204,6 +204,14 @@ export async function deleteNode({ commit, dispatch }, id) {
   }
 }
 
+export async function exportTapestry({ dispatch }) {
+  try {
+    return await client.exportTapestry()
```

Comment:
You don't need the `async` and `await` here since `client.exportTapestry()` already returns a Promise. Also would probably be a good idea to rename this action too (as above).

```suggestion
export function exportTapestry({ dispatch }) {
  try {
    return client.exportTapestry()
```

Reply:
@narendrasss I think because it's nested in a try-catch block, without await, the catch block won't be reached and it will just return the promise as received, but with await, a rejected promise will throw an error that can be caught